packages/client/hmi-client/src/workflow/ops/calibrate-julia/tera-calibrate-julia.vue

```diff
@@ -172,15 +203,34 @@ const modelConfig = ref<ModelConfiguration>();
 const modelConfigId = computed<string | undefined>(() => props.node.inputs[0]?.value?.[0]);
 const datasetId = computed<string | undefined>(() => props.node.inputs[1]?.value?.[0]);
 const currentDatasetFileName = ref<string>();
-const simulationIds = computed<any | undefined>(() => props.node.outputs[0]?.value);
-const runResults = ref<RunResults>({});
-const parameterResult = ref<{ [index: string]: any }[]>();
 const mapping = ref<CalibrateMap[]>(props.node.state.mapping);
 
+const {
+	runResults,
+	selectedRun,
+	currentIntermediateVals,
+	parameterResult,
+	showSpinner,
+	runInProgress
+} = useJuliaCalibrateOptions();
+const runList = computed(() =>
+	Object.keys(props.node.state.calibrateConfigs.runConfigs).map((runId: string, idx: number) => ({
+		label: `Output ${idx + 1} - ${runId}`,
+		runId
+	}))
+);
+const drilldownLossPlot = ref<HTMLElement>();
+
+const currentParams = computed(() =>
+	showSpinner.value
+		? parameterResult.value
+		: props.node.state.calibrateConfigs.runConfigs[selectedRun.value?.runId]?.params
+);
```

Comment:
We know when the calibration finishes, so we don't need this kind of logic - also please avoid predicating logic on the state of the UI.

Reply:
removed